src/test/java/org/sagebionetworks/bridge/services/HealthDataDocumentationServiceTest.java

```diff
@@ -0,0 +1,134 @@
+package org.sagebionetworks.bridge.services;
+
+import com.google.common.collect.ImmutableList;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.sagebionetworks.bridge.dao.HealthDataDocumentationDao;
+import org.sagebionetworks.bridge.exceptions.BadRequestException;
+import org.sagebionetworks.bridge.exceptions.InvalidEntityException;
+import org.sagebionetworks.bridge.models.ForwardCursorPagedResourceList;
+import org.sagebionetworks.bridge.models.HealthDataDocumentation;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.sagebionetworks.bridge.BridgeConstants.API_DEFAULT_PAGE_SIZE;
+import static org.sagebionetworks.bridge.TestConstants.IDENTIFIER;
+import static org.sagebionetworks.bridge.TestConstants.TEST_APP_ID;
+import static org.sagebionetworks.bridge.models.ResourceList.OFFSET_KEY;
+import static org.testng.AssertJUnit.assertSame;
+
+public class HealthDataDocumentationServiceTest {
+
+    @Mock
+    private HealthDataDocumentationDao mockDao;
+
+    @InjectMocks
+    private HealthDataDocumentationService service;
+
+    @BeforeMethod
+    public void before() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void createOrUpdateHealthDataDocumentation() {
+        HealthDataDocumentation doc = makeValidDoc();
+        when(mockDao.createOrUpdateDocumentation(doc)).thenReturn(doc);
+
+        HealthDataDocumentation result = service.createOrUpdateHealthDataDocumentation(doc);
+        assertSame(result, doc);
+
+        verify(mockDao).createOrUpdateDocumentation(doc);
+    }
```

Comment:
We should also verify that modifiedOn and modifiedBy to the result.

You can mock DateTime.now() by calling DateTimeUtils.setCurrentMillisFixed(). After the test, you should call DateTimeUtils.setCurrentMillisSystem() to reset the clock. See for example https://github.com/Sage-Bionetworks/BridgeServer2/blob/develop/src/test/java/org/sagebionetworks/bridge/upload/InitRecordHandlerTest.java#L43

For RequestContext, you call RequestContext.set(). After the test, you can RequestContext.set(null) to reset the RequestContext. See for example https://github.com/Sage-Bionetworks/BridgeServer2/blob/develop/src/test/java/org/sagebionetworks/bridge/validators/ExternalIdValidatorTest.java#L36